### PR TITLE
Display PostgreSQL server created in resource

### PR DESCRIPTION
**Pull request by pamelafox**
Currently, azd doesn't display when it creates a PostgreSQL flexible server, but it should, since it displays SQL server creations.

I based this PR off https://github.com/Azure/azure-dev/pull/1320/files

### CODE REVIEW

File: cli/azd/pkg/infra/azure_resource_types.go [hunk 1]
```diff
@@ -64,10 +65,11 @@ func GetResourceTypeDisplayName(resourceType AzureResourceType) string {
 		return "Azure API Management"
 	case AzureResourceTypeCacheForRedis:
 		return "Cache for Redis"
-	case AzureResourceTypeSqlDatabase:
-		return "Azure SQL DB"
+	case AzureResourceTypeSqlServer:
+		return "Azure SQL Server"
+	case AzureResourceTypePostgreSqlServer:
+		return "Azure PostgreSQL Server"
```
Comment (pamelafox) [inline cli/azd/pkg/infra/azure_resource_types.go:71]: There's also a "Standard server" so I could have specified "Flexible server" here, but from what I understand, standard servers are on the way out so I don't know that azd needs to support them. (i.e. flexible server new hotness)

**Comment by pamelafox** [comment]
I have not tested this since I haven't succeeded on getting azd running on my machine, but if the CI runs that creates the build artifact, I can test with that.

**Review by wbreza — APPROVED** [review]

**Comment by pamelafox** [comment]
Successful deploy with PostgreSQL flexible server:

Screenshot of azd up

Successful deploy with SQL server:

Screenshot of azd up

**Comment by pamelafox** [comment]
Hm, I just realized that only the recently added resources have "Azure" in front of their textual representation, the rest leave it out. I don't know whether none should have "Azure" for consistency, or if there's a reason why some start with "Azure".

**Comment by weikanglim** [comment]
@pamelafox Try to stick to whatever the Azure Portal shows the resource name as. That is where we'll move towards longer terms once we get ahold of the proper translation assets from Azure Portal (and eliminate the manually defined translations instead).

File: cli/azd/pkg/infra/azure_resource_types.go (hunk 1 above)
Comment (weikanglim) [reply cli/azd/pkg/infra/azure_resource_types.go:71]: Don't really have a strong opinion -- can go either way, but "Azure Database for PostgreSQL flexible server" is the translation provided by Portal currently.

Comment (weikanglim) [reply cli/azd/pkg/infra/azure_resource_types.go:71]: Which is a real mouthful but that's what the service owners decided on..

**Review by weikanglim — APPROVED** [review]

**Comment by pamelafox** [comment]
@weikanglim Okay, updated to the Portal name. It's a mouthful but sounds like that's what it'll be in the future too. Thanks for clarifying!

**Review by weikanglim — APPROVED** [review]

**Pull request merged**